packages/compile-solidity/test/test_nonrelative.js

```diff
@@ -59,3 +62,58 @@ describe("Non-relative non-absolute file paths", function () {
     process.chdir(originalWorkingDirectory);
   });
 });
+
+describe("Non-canonical absolute file paths", function () {
+  this.timeout(5000); // solc
+  let tmpdir;
+  let options;
+
+  before("Set up temporary directory and project", async function () {
+    tmpdir = tmp.dirSync({ unsafeCleanup: true }).name; //tmp uses callbacks, not promises, so using sync
+    await fs.promises.mkdir(path.join(tmpdir, "./contracts"));
+    const contracts_directory = path.join(tmpdir, "./contracts");
+    options = {
+      working_directory: tmpdir,
+      contracts_directory,
+      contracts_build_directory: path.join(tmpdir, "./build/contracts"), //nothing is actually written, but resolver demands it
+      compilers: {
+        solc: {
+          version: "0.8.6",
+          settings: {
+            optimizer: {
+              enabled: false,
+              runs: 200
+            }
+          }
+        }
+      },
+      quiet: true
+    };
+    options.resolver = new Resolver(options);
+    const importedPath = path.join(options.contracts_directory, "Imported.sol");
+    const importerPath = path.join(options.contracts_directory, "DoubleSlash.sol");
+    await fs.promises.copyFile(
+      path.join(__dirname, "./sources/badSources/Imported.sol"),
+      importedPath
+    );
+    await fs.promises.writeFile(
+      importerPath,
+      `import "/${importedPath}";` //note the extra slash
```

Comment:
This is certainly not a blocker for this PR. 

I just want to mention we should have a categorization method that selects tests for the current platform. In this test, `/drive:\Users\... is not a valid Windows path`. Perhaps tags can work, or we can inject platform specific guards in the tests. Something to discuss with the team I guess.

Reply:
Yeah I think that might be better addressed on #4206?

Reply:
Yes, of course.